File: app/src/main/java/com/channa/mobiledatausageapp/MyApplication.java
package com.channa.mobiledatausageapp;

import android.app.Application;

import com.channa.mobiledatausageapp.di.component.ApplicationComponent;
import com.channa.mobiledatausageapp.di.component.DaggerApplicationComponent;
import com.channa.mobiledatausageapp.di.module.ApplicationContextModule;
import com.channa.mobiledatausageapp.di.module.DataModule;


public class MyApplication extends Application {

    private ApplicationComponent applicationComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        applicationComponent = DaggerApplicationComponent
                .builder()
                .applicationContextModule(new ApplicationContextModule(this))
                .dataModule(new DataModule())
                .build();

    }

    public ApplicationComponent getApplicationComponent() {
        return applicationComponent;
    }
}

File: app/src/main/java/com/zak/datauses/MyApplication.kt
package com.zak.datauses

import android.app.Application
import com.zak.datauses.di.component.ApplicationComponent
import com.zak.datauses.di.component.DaggerApplicationComponent
import com.zak.datauses.di.module.ApplicationContextModule
import com.zak.datauses.di.module.DataModule

class MyApplication : Application() {
    private var applicationComponent: ApplicationComponent? = null
    override fun onCreate() {
        super.onCreate()
        applicationComponent = DaggerApplicationComponent
                .builder()
                .applicationContextModule(ApplicationContextModule(this))
                .dataModule(DataModule())
                .build()
    }

    fun getApplicationComponent(): ApplicationComponent? {
        return applicationComponent
    }
}